### ISSUE TITLE
XmlPoke calls the wrong function?

### ISSUE THREAD

**Issue by AnderssonPeter** [cake-build/cake#765]
the following code
```
var webConfigFile = File(@"Web.config");
if (FileExists(webConfigFile))
    Information("Found file!");
XmlPoke(webConfigFile, "/configuration/appSettings/add[@key = 'ApplicationName']/@value", "Test");
```
Calls `XmlPoke(string, ​string, ​string)` it should call `XmlPoke(FilePath, ​string, ​string)`.

If any one else has this issue just cast you `File` to a `FilePath` befor passing it to `XmlPoke`.
`XmlPoke((FilePath)webConfigFile, .....`.

**Comment by brentpi** [comment]
Currently encountering this as well.  File exists, but when attempting to parse, produces "Error: Data at the root level is invalid. Line 1, position 1.".  This fix worked.

**Comment by devlead** [comment]
This was addressed by #978
